sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1669,6 +1652,77 @@ public boolean apply(@Nonnull String input) {
     return convertedOptions;
   }
 
+  /**
+   * Returns true if the given type is a SIMPLE_TYPES, enum or any of these types in a
+   * parameterized ValueProvider.
+   */
+  private static boolean isSimpleType(Class<?> type, JavaType genericType) {
+    Class<?> unwrappedType = type.equals(ValueProvider.class)
+        ? genericType.containedType(0).getRawClass() : type;
+    return SIMPLE_TYPES.contains(unwrappedType) || unwrappedType.isEnum();
+  }
+
+  /**
+   * Returns true if the given type is a any Array or Collection of SIMPLE_TYPES or enum, or
+   * any of these types in a parameterized ValueProvider.
+   */
+  private static boolean isCollectionOrArrayOfAllowedTypes(Class<?> type, JavaType genericType) {
+    Class<?> containerType = type.equals(ValueProvider.class)
+        ? genericType.containedType(0).getRawClass() : type;
+
+    // Check if it is an array of simple types or enum.
+    if (containerType.isArray() && (SIMPLE_TYPES.contains(containerType.getComponentType())
+        || containerType.getComponentType().isEnum())) {
+        return true;
+    }
+    // Check if it is Collection of simple types or enum.
+    if (Collection.class.isAssignableFrom(containerType)) {
+      JavaType innerType = type.equals(ValueProvider.class)
+          ? genericType.containedType(0).containedType(0)
+          : genericType.containedType(0);
+      // Note that raw types are allowed, hence the null check.
+      if (innerType == null || SIMPLE_TYPES.contains(innerType.getRawClass())
+          || innerType.getRawClass().isEnum()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Ensures that empty string value is allowed for a given type.
+   *
+   * <p>Empty strings are only allowed for String, String Array, Collection of Strings or any of
+   * these types in a parameterized ValueProvider.
+   *
+   * @param type class object for the type under check.
+   * @param genericType complete type information for the type under check.
+   * @param genericTypeName a string representation of the complete type information.
+   */
+  private static void checkEmptyStringAllowed(Class<?> type, JavaType genericType,
+      String genericTypeName) {
+    Class<?> unwrappedType = type.equals(ValueProvider.class)
+        ? genericType.containedType(0).getRawClass() : type;
+
+    Class<?> containedType = unwrappedType;
+    if (unwrappedType.isArray()) {
+      containedType = unwrappedType.getComponentType();
+    } else if (Collection.class.isAssignableFrom(unwrappedType)) {
+      JavaType innerType = type.equals(ValueProvider.class)
```

Comment:
Ditto on `Collection<ValueProvider<String>>` here not making sense.
`ValueProvider<Collection<String>>` makes total sense.